service/matching/physical_task_queue_manager.go

```diff
@@ -642,8 +642,10 @@ func (c *physicalTaskQueueManagerImpl) ensureRegisteredInDeploymentVersion(
 		var errTooMany workerdeployment.ErrMaxTaskQueuesInDeployment
 		if errors.As(err, &errTooMany) {
 			c.deploymentRegisterError = errTooMany
+			return errTooMany
 		}
-		return err
+		c.logger.Error("error while registering version", tag.Error(err))
```

Comment:
also would be nice to have the version string in this error message, although maybe it's already in the wrapped error